ddtrace/contrib/internal/asgi/middleware.py

```diff
@@ -34,6 +34,7 @@
         request_span_name="asgi.request",
         distributed_tracing=True,
         _trace_asgi_websocket=os.getenv("DD_ASGI_TRACE_WEBSOCKET", default=False),
+        _obfuscate_404_resource=os.getenv("DD_ASGI_OBFUSCATE_404_RESOURCE", default=False),
```

Comment:
nit: If we wanted to make this more general, we could accept a list of status codes (even if it defaults to 404) to apply this logic to, in case this issue occurs for 5xx or any other status code beyond 404.